### PR TITLE
Add back reference matcher.

### PR DESCRIPTION
**Pull request by vwxyzh**

### CODE REVIEW

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/BackRefereneceMatcher.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.DocAsCode.MarkdownLite.Matchers
+{
+    internal sealed class BackRefereneceMatcher : Matcher
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/BackRefereneceMatcher.cs]: >Referenece [](start = 30, length = 10)

typo: Reference

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs [hunk 2]
```diff
@@ -183,5 +193,118 @@ private static void ValidateMatcherArray(Matcher[] matchers)
                 }
             }
         }
+
+        /// <summary>
+        /// Sequence.
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs]: >Sequence [](start = 12, length = 8)

?

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs [hunk 3]
```diff
@@ -183,5 +193,118 @@ private static void ValidateMatcherArray(Matcher[] matchers)
...
+        /// </summary>
+        public static Matcher operator +(Matcher left, Matcher right)
+        {
+            var seqLeft = left as SequenceMatcher;
+            var seqRight = right as SequenceMatcher;
+            if (seqLeft != null)
+            {
+                if (seqRight != null)
+                {
+                    return Sequence(seqLeft.Inners.Concat(seqRight.Inners).ToArray());
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs]: >ToArray [](start = 75, length = 7)

is IEnumerable enough for Sequence ctor?

**Comment by hellosnow** [comment]
:shipit:

File: src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs [hunk 4]
```diff
@@ -183,5 +193,118 @@ private static void ValidateMatcherArray(Matcher[] matchers)
...
+        }
+
+        /// <summary>
+        /// Repeat.
+        /// </summary>
+        public static Matcher operator *(Matcher matcher, int count) =>
+            Repeat(matcher, count, count);
+
+        /// <summary>
+        /// Repeat.
```
Comment (vicancy) [inline src/Microsoft.DocAsCode.MarkdownLite/Matchers/Matcher.cs]: >Repeat [](start = 12, length = 6)

... more details?

**Comment by vicancy** [comment]
:shipit:

**Pull request merged**